src/GridFS/StreamWrapper.php

```diff
@@ -122,14 +143,44 @@ public function stream_eof(): bool
      */
     public function stream_open(string $path, string $mode, int $options, ?string &$openedPath): bool
     {
-        $this->initProtocol($path);
+        $protocol = $this->initProtocol($path);
+
+        assert(is_resource($this->context));
+        $contextOptions = stream_context_get_options($this->context)[$protocol] ?? null;
+
+        if ($contextOptions === null) {
+            if (! isset(self::$contextResolver)) {
+                if ($options & STREAM_REPORT_ERRORS) {
```

Comment:
TIL this constant exists. Looks like it was introduced a [very long time ago](https://github.com/php/php-src/commit/32a4f35580436c0c7e74499b78f2390ab7fe367b#diff-aa9fffa7bf1ce5999dd7fb28a1dc1cb2d3841f97880dbde51d5b12a109669699R54).

This seems like a change we should track in a separate ticket for changelog visibility.

Reply:
I'm not sure we can pass this option with the current code, because `fopen` cannot be used outside of the Bucket class. So I d'ont think this could impact current users. but I may be wrong as I didn't fully understood how this option is set.

Tracked in PHPLIB-1318.

Reply:
This should not have any impact on users since streams are only created internally in PHPLIB. If you plan on changing that, then I expect you'd want to handle this option.